components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java

```diff
@@ -91,7 +92,13 @@ public boolean process(Exchange exchange, AsyncCallback async) {
         final Boolean await = msg.getHeader(MiloConstants.HEADER_AWAIT, this.defaultAwaitWrites, Boolean.class);
 
         if (TRUE.equals(await)) {
-            future.whenComplete((v, ex) -> async.done(false));
+            future.whenComplete((o, throwable) -> {
+                        final DefaultMessage newMessage = new DefaultMessage(exchange);
```

Comment:
I dont think you need to set a new message as this leads to loosing headers and whatnot.

You usually just set the result directly on the message, eg

exchange.getMessage().setBody(...)